packages/dds/cell/src/cell.ts

```diff
@@ -158,6 +237,28 @@ export class SharedCell<T = any> extends SharedObject<ISharedCellEvents<T>> impl
         return this.data === undefined;
     }
 
+    /**
+     * {@inheritDoc ISharedCell.getAttribution}
+     */
+    public getAttribution(key: AttributionKey): AttributionInfo {
+        return this.attributor[key.seq];
+    }
+
+    /**
+     * {@inheritDoc ISharedCell.setAttribution}
+     */
+    public setAttribution(message: ISequencedDocumentMessage): void {
+        this.attributionKey = { type: "op", seq: message.sequenceNumber };
+        this.attributor[message.sequenceNumber] = { user: { id: message.clientId }, timestamp: message.timestamp };
```

Comment:
@Abe27342 Considering the attributor is just a single-state in cell, so I used key-value pair structure to store the `attributionKey -> attributionInfo`, where `attributionKey` is consistent with the design in merge-tree (`{type: 'op', seq: number}`) But It seems not support use the `attributionKey` as the key, so I still use the seq number instead. Does this design look weird?

Reply:
i think you don't need an object at all. You can just have a single `AttributionKey`. `SharedCell` only has one piece of data at a time, and the current design for attribution is not targeted on supporting history-like features.

`private attribution: AttributionKey | undefined;` in the class's state and then

`this.attribution = { type: "op", seq: message.sequenceNumber }` on this line.